src/Microsoft.AspNetCore.Localization/QueryStringRequestCultureProvider.cs

```diff
@@ -71,9 +76,34 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 queryCulture = queryUICulture;
             }
 
+            if (!IsValidCulture(queryCulture))
+            {
+                _logger = _logger ?? httpContext.RequestServices.GetService<ILogger<QueryStringRequestCultureProvider>>();
```

Comment:
I tried that. Unfortunately we don't have an `AddLocalization()` to hook up the options setup to this.